### PR TITLE
Update the affine parameter description in several classes

### LINKED ISSUE #3317: unexpected chaining of `affine` and `scale` parameters in Viewer.add_image
## 🐛 Bug

When calling the `add_image` method of a Viewer, if both the `affine` as well as `scale`,`translation` and/or `shear` are specified the effects seem to be additive. This disagrees with the description in the `add_image` docstring:

https://github.com/napari/napari/blob/8f1126c82bfbabd6cafd9739fa8c747ddff79583/napari/components/viewer_model.py#L652-L653

(the description in the `_ImageBase` class also [says the same](https://github.com/napari/napari/blob/f1b3be15b33ddd141575d56dd315f5f5676c1d94/napari/layers/image/image.py#L93-L94))

After looking a bit at the code, it seems that the `Layer` class does indeed chain these transformations:
https://github.com/napari/napari/blob/f1b3be15b33ddd141575d56dd315f5f5676c1d94/napari/layers/base/base.py#L254-L268

And notably the `affine` parameter description on the `Layer` class does describe the current behavior accurately:
https://github.com/napari/napari/blob/f1b3be15b33ddd141575d56dd315f5f5676c1d94/napari/layers/base/base.py#L144-L145

Obviously one can work around this by only supplying one or the other so that there is not redundant information.

## To Reproduce

I noticed this when playing around with these arguments with a development version of the `napari.nibabel` plugin. If I set both, the slices from a single slice MRI scan appear much thicker in the slice direction than one would expect from the affine alone.

## Expected behavior

It sounds like we most likely just need to update some docstrings to match the current behavior. Is this correct? If so, I can make a PR for that.

## Environment

current Napari development branch (~0.4.11rc2)

### PR DESCRIPTION
**Pull request by grlee77**
# Description

closes #3317

As discussed in #3317, the `affine` parameter description in some classes describes outdated behavior. This PR is to make it consistently describe that the affine transform will be chained with any scale, translation or shear that is provided.

## Type of change
- [x] This change requires a documentation update

# References
inspection of `Layer.__init__` to verify that the transforms are chained.

# How has this been tested?
manual inspection that the changed classes pass the affine on to Layer via `super.__init__()` and `Layer` does chain the transforms. 

This was also verified manually in the GUI canvas when displaying an `Image` added via `Viewer.add_image`

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
~- [ ] I have commented my code, particularly in hard-to-understand areas~
- [x] I have made corresponding changes to the documentation
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: napari/layers/intensity_mixin.py
Comment (andy-sweet) [inline napari/layers/intensity_mixin.py:21]: Another good catch!

Nit: `Image` is subclass of `Layer` that we define, but its declaration is `class Image(_ImageBase)` where we have `class _ImageBase(IntensityVisualizationMixIn, Layer)`, so this might be confusing. But I guess this is just an example, so this `Image` does not have to correspond to our `Image`, so you can also ignore me here.

**Review by andy-sweet — APPROVED** [review]
Thanks for catching these.

After seeing how many times these are repeated, I think that we might want to consider doing something like numpy (and probably others) do when they have repeated kwargs in many places. For example, the docs for [`numpy.add`](https://numpy.org/doc/stable/reference/generated/numpy.add.html) redirect the reader to general the `ufunc` docs. In the same way, we could redirect to the `Layer` docs. What do you, and others, think?

My guess is that this may not be the only parameter docstring that's out of sync, and almost certainly won't be the last (especially as I continue to make contributions...).

Anyway, this is an improvement over what we have, so I approve. But you should also get one of the core devs to approve and maybe we can run the idea above past them.

File: napari/layers/intensity_mixin.py
Comment (grlee77) [reply napari/layers/intensity_mixin.py:21]: Right, happy to change it to `_ImageBase` if that is preferred

**Comment by grlee77** [comment]
> After seeing how many times these are repeated, I think that we might want to consider doing something like numpy (and probably others) do when they have repeated kwargs in many places. For example, the docs for numpy.add redirect the reader to general the ufunc docs. In the same way, we could redirect to the Layer docs. What do you, and others, think?

`scipy.ndimage` has a special decorator that can replace variables with centrally defined parameter text. For example, [this example scipy.ndimage docstring](https://github.com/scipy/scipy/blob/11505df087ae68ebe083e937d162d351374d80f8/scipy/ndimage/_filters.py#L83-L100) and others get the actual text [as defined here](https://github.com/scipy/scipy/blob/11505df087ae68ebe083e937d162d351374d80f8/scipy/ndimage/_ni_docstrings.py). The [doccer.py](https://github.com/scipy/scipy/blob/master/scipy/_lib/doccer.py) utility that enables this is not part of the public API, but we could vendor it if we wanted to use this kind of approach.

This makes sure things stay consistent across functions, but a downside is that it is a little bit annoying when reading the source code directly. You have to go reference the text in that separate file or in the generated documentation rather than seeing it right inline.

**Review by sofroniewn — APPROVED** [review]
LGTM! Thanks!!

File: napari/layers/intensity_mixin.py
Comment (andy-sweet) [reply napari/layers/intensity_mixin.py:21]: Well, `_ImageBase` is protected, so maybe it's actually worse. You could consider `Surface` as that (at least currently) does immediately inherit `IntensityVisualizationMixin`. Or you can just leave this as is. I'll leave it up to you.

**Comment by andy-sweet** [comment]
> `scipy.ndimage` has a special decorator that can replace variables with centrally defined parameter text. For example, [this example scipy.ndimage docstring](https://github.com/scipy/scipy/blob/11505df087ae68ebe083e937d162d351374d80f8/scipy/ndimage/_filters.py#L83-L100) and others get the actual text [as defined here](https://github.com/scipy/scipy/blob/11505df087ae68ebe083e937d162d351374d80f8/scipy/ndimage/_ni_docstrings.py). The [doccer.py](https://github.com/scipy/scipy/blob/master/scipy/_lib/doccer.py) utility that enables this is not part of the public API, but we could vendor it if we wanted to use this kind of approach.
> 
> This makes sure things stay consistent across functions, but a downside is that it is a little bit annoying when reading the source code directly. You have to go reference the text in that separate file or in the generated documentation rather than seeing it right inline.

That's pretty neat, though I was thinking of something simpler that would just use a reST directive like the following.

```
For other keyword only arguments see :class:`napari.layers.Layer`.
```

Sphinx will then generate the link in the built docs, but if you're in the source code, you'll still have to manually navigate to `Layer` from `Points` (unless there are some reST/sphinx plugins for some IDEs that make it a clickable link), though it's fairly clear where to find it.

Anyway, this is definitely a future discussion (maybe part of the documentation working group?), and this PR should be submitted as is.

**Pull request merged**
